src/main/java/net/snowflake/client/jdbc/SnowflakeConnectionV1.java

```diff
@@ -424,18 +424,14 @@ public Statement createStatement(int resultSetType, int resultSetConcurrency)
   @Override
   public Statement createStatement(
       int resultSetType, int resultSetConcurrency, int resultSetHoldability) throws SQLException {
-    if (isUsingDefaultConnectionHandler) {
```

Comment:
Here, we remove the check that we had previously because we didn't have this piece in the `SFConnectionHandler` interface yet; we don't need it any more because we pass the `SnowflakeConnection` object to `SnowflakeStatementV1`, which constructs an `SFBaseStatement` by invoking the `getSFStatement()` method on the `SFConnectionHandler`.